src/Pact/Eval.hs

```diff
@@ -649,11 +708,21 @@ reduce (TObject (Object ps t ko oi) i) =
 reduce (TBinding ps bod c i) = case c of
   BindLet -> reduceLet ps bod i
   BindSchema _ -> evalError i "Unexpected schema binding"
-reduce t@TModule{} = evalError (_tInfo t) "Modules and Interfaces only allowed at top level"
+reduce TModule{..} = TModule
+  <$> traverse reduce _tModuleDef
+  <*> return (abstract (const Nothing) $ TLiteral (LString "hack redux") def)
+  <*> return _tInfo
 reduce t@TUse {} = evalError (_tInfo t) "Use only allowed at top level"
 reduce t@TStep {} = evalError (_tInfo t) "Step at invalid location"
 reduce TSchema {..} = TSchema _tSchemaName _tModule _tMeta <$> traverse (traverse reduce) _tFields <*> pure _tInfo
 reduce TTable {..} = TTable _tTableName _tModuleName _tHash <$> mapM reduce _tTableType <*> pure _tMeta <*> pure _tInfo
+reduce t@TDynamic {} = evalError (_tInfo t)
```

Comment:
Hmmm ... this might represent an interface `TConst` ... what could we do here if we did want to try to reduce?